File: gaia-sdk-javascript/src/graphql/request/type/BehaviourExecution.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents behaviour execution information
 */
export class BehaviourExecution extends Array<(_:VariableRegistry) => string> {
public _typeName = "BehaviourExecution";
    public processInstanceId = () => { 
        this.push(_ => "processInstanceId")
    };

    public identityId = () => { 
        this.push(_ => "identityId")
    };

    public state = () => { 
        this.push(_ => "state")
    };

    public name = () => { 
        this.push(_ => "name")
    };

    public duration = () => { 
        this.push(_ => "duration")
    };

    public behaviourId = () => { 
        this.push(_ => "behaviourId")
    };

    public created = () => { 
        this.push(_ => "created")
    };

    public updated = () => { 
        this.push(_ => "updated")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/BehaviourExecution.py
from gaia_sdk.api.VariableRegistry import VariableRegistry


class BehaviourExecution(list):
    """
    Represents behaviour execution information
    """

    def process_instance_id(self):
        self.append(lambda x: "processInstanceId")

    def identity_id(self):
        self.append(lambda x: "identityId")

    def state(self):
        self.append(lambda x: "state")

    def name(self):
        self.append(lambda x: "name")

    def duration(self):
        self.append(lambda x: "duration")

    def behaviour_id(self):
        self.append(lambda x: "behaviourId")

    def created(self):
        self.append(lambda x: "created")

    def updated(self):
        self.append(lambda x: "updated")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
